test/ability_sap_sipper.c

```diff
@@ -0,0 +1,54 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Sap Sipper negates damage from Grass-type moves")
+{
+    GIVEN {
+        PLAYER(SPECIES_MARILL) { Ability(ABILITY_SAP_SIPPER); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_VINE_WHIP); }
+    } SCENE {
+        NONE_OF { HP_BAR(player); }
```

Comment:
Let's add the canceling move messages at least.

Reply:
Then let's add the regular messages to make it explicit how the messaging shows up. I had to look up a video to make sure.